File: fe-modul-report/src/features/dashboard/types/targetReport.ts
export interface TargetReportRequest {
  id?: string | null;
  date: string; // Format: YYYY-MM-DD
  targetId: string;
  productionOrderId?: number | null;
  productionDate: string; // Format: YYYY-MM-DD
  totalDays?: number | null;
  targetPerDay?: number | null;
  shiftDone: number; // caDaThiCong
  shiftPlus?: number | null;
  shiftRemain?: number | null;
  targetPerDayRemain?: number | null;
  performDone: number; // thucHienNgay
  monthLyCumulative?: number | null;
  donePercent?: number | null;
}

export interface TargetReportResponse {
  id: string;
  targetId: string;
  targetName: string;
  code: string;
  unit: string;
  value: number; // monthly target value
  productionOrderId?: number | null;
  productionDate: string; // Format: YYYY-MM-DD
  totalDays?: number | null;
  targetPerDay?: number | null;
  shiftDone?: number | null;
  shiftPlus?: number | null;
  shiftRemain?: number | null;
  targetPerDayRemain?: number | null;
  performDone?: number | null;
  monthLyCumulative?: number | null;
  donePercent?: number | null;
  children?: TargetReportResponse[];
}

export interface DepartmentTargetResponse {
  departmentId: string;
  departmentName: string;
  date: string; // Format: YYYY-MM-DD
  targetReportResponseList: TargetReportResponse[];
  children?: DepartmentTargetResponse[];
}

File: be/src/main/java/com/quangnt0000/be_modul/modal/DataLake/TargetReport.java
package com.quangnt0000.be_modul.modal.DataLake;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;
import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
@FieldDefaults(level = lombok.AccessLevel.PRIVATE)
public class TargetReport {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String id;

    LocalDate date;

    @ManyToOne
    @JoinColumn(name = "target_id")
    private Target target;

    Integer productionOrderId;
    LocalDate productionDate;

    Integer totalDays;

    BigDecimal targetPerDay;

    Integer shiftDone;
    Integer shiftPlus;
    Integer shiftRemain;

    BigDecimal targetPerDayRemain;
    BigDecimal performDone;

    BigDecimal monthLyCumulative;
    BigDecimal donePercent;

    @Builder.Default
    Boolean deleted = false;
}
